keylime/tpm/tpm2_objects.py

```diff
@@ -61,22 +61,30 @@ def _pack_in_tpm2b(val):
 
 
 class NonAsymAlgSpecificParameters:
-    sym_algorithm = None
-    sym_keybits = None
-    sym_mode = None
-    sym_details = None
-    scheme_scheme = None
-    scheme_details = None
-
-    def __init__(self, sym_algorithm, sym_keybits, sym_mode, sym_details, scheme_scheme, scheme_details):
+    sym_algorithm: Optional[int] = None
+    sym_keybits: Optional[int] = None
+    sym_mode: Optional[int] = None
+    sym_details: Optional[int] = None
+    scheme_scheme: Optional[int] = None
+    scheme_details: Optional[int] = None
```

Comment:
IMHO this seems to be a confusion about class fields and instance fields? If so should be dropped.

Reply:
These can at least serve as type specifiers if we modify them to something like this:
```
    sym_algorithm: int
    sym_keybits: int
    sym_mode: int
    sym_details: int
    scheme_scheme: int
    scheme_details: int
```

When I change sym_algorithm to be 'str' then mypy complains like this:

```
keylime/tpm/tpm2_objects.py:80: error: Incompatible types in assignment (expression has type "int", variable has type "str")  [assignment]
```

So there's a point in having these but initializing them is wrong, which makes them class variables.
I introduced such type specifiers in other parts as well without initializing them... and we already had them.

Reply:
Sure, I was pointing to the original code.